numba/ir.py

```diff
@@ -186,6 +198,54 @@ def with_lineno(self, line, col=None):
 unknown_loc = Loc("unknown location", 0, 0)
 
 
+@total_ordering
+class SlotEqualityCheckMixin(object):
+    # some ir nodes are __dict__ free using __slots__ instead, this mixin
+    # should not trigger the unintended creation of __dict__.
+    __slots__ = tuple()
+
+    def __eq__(self, other):
+        if type(self) == type(other):
+            for name in self.__slots__:
```

Comment:
Note that this will not support inheritance.  An instance's `__slots__` can only see the version defined in its type but not it's parent's.